watchmaker/managers/base.py

```diff
@@ -191,40 +211,49 @@ def download_file(self, url, filename, sourceiss3bucket=False):
             except Exception as exc:
                 # TODO: Update `except` logic
                 logging.error('Unable to download file from web server.\n'
-                                  'url = {0}\n'
-                                  'filename = {1}\n'
-                                  'Exception: {2}'
-                                  .format(url, filename, exc))
+                              'url = {0}\n'
+                              'filename = {1}\n'
+                              'Exception: {2}'
+                              .format(url, filename, exc))
                 raise SystemError('Unable to download file from web server.\n'
                                   'url = {0}\n'
                                   'filename = {1}\n'
                                   'Exception: {2}'
                                   .format(url, filename, exc))
             logging.debug('Downloaded file from web server -- \n'
-                  '    url      = {0}\n'
-                  '    filename = {1}'.format(url, filename))
+                          '    url      = {0}\n'
+                          '    filename = {1}'.format(url, filename))
+
+    def call_process(self, cmd):
+        if not isinstance(cmd, list):
+            exceptionhandler('Command is not a list.\n{0}'.format(str(cmd)))
+        rsp = subprocess.call(cmd)
+
+        if rsp != 0:
```

Comment:
I dont think that will work.  Since our exceptionhandler uses sys.exit(), i dont think the exception block with execute since the process will die before a return.  If you want to be able to handle a failed call, we will need to use a flag to redirect the exception.  (i think this was your idea when we were talking about it earlier)  Something like this:

```
	def call_process(self, cmd, returnexception=False):
        if not isinstance(cmd, list):
            exceptionhandler('Command is not a list.\n{0}'.format(str(cmd)))
        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
        out, err = process.communicate()
        if process.returncode != 0 and returnexception:
            return subprocess.CalledProcessError, err
        elif process.returncode !=0:
            exceptionhandler('Command execution failed.\n{0}\nError Message:{1}'.format(cmd, err)
        else:
            logging.info(out)
```